cql3/expr/expression.cc

```diff
@@ -741,6 +741,34 @@ expression make_conjunction(expression a, expression b) {
     return conjunction{std::move(children)};
 }
 
+static
+void
+do_factorize(std::vector<expression>& factors, expression e) {
+    // A TRUE factor can be eliminated
```

Comment:
It can, but do we want to do this here? (genuine question). The name doesn't suggest that it also does an optimization pass, so it may be theoretically undesirable for (TRUE AND TRUE) to be factorized to (), and being equal to factorized (TRUE). And since it also produces empty vectors in certain cases, we need to make sure that the code is ready for it. It probably is and in general it's a good idea to optimize early, I'm just noting that it may be a surprise for callers.

Reply:
I was anticipating that single-expression WHERE would require me to pass TRUE to select_statement's constructor, and this would remove it back. It doesn't, so we probably pass an empty conjunction somewhere.

Ah, it's quite by accident - after removing `bool` as a variant member (ad3d2ee47d00ea94818054ff138476172e17ac36), the first element of the variant is conjunction, so the default expression constructor creates an empty conjunction, and if the WHERE clause doesn't exist, that's what we get.

I can remove the TRUE factoring for now. I expect we'll add it back, but we can add it in a different place.